nmdc_schema/migrators/partials/migrator_from_11_0_3_to_11_1_0/migrator_from_11_0_3_to_11_1_0_part_1.py

```diff
@@ -0,0 +1,217 @@
+        >>> db = {
+        ...   'data_object_set': [
+        ...     {'id': 'nmdc:dobj-13-abc123'},
+        ...     {'id': 'nmdc:dobj-13-def456'}
+        ...   ]
+        ... }
+        >>> a = DictionaryAdapter(database=db)
+        >>> m = Migrator(adapter=a)
+        >>> m.check_for_valid_data_object('nmdc:dobj-13-abc123')  # Exists in database
+        True
+        >>> m.check_for_valid_data_object('nmdc:dobj-13-nonexistent')  # Doesn't exist
+        False
+        """
+
+        data_obj_doc = self.adapter.get_document_having_value_in_field(
+                    collection_name="data_object_set", field_name="id", value=data_obj_id
+                    )
+        
+        return data_obj_doc is not None
+
+    def store_and_remove_calibrations(self, workflow_execution_doc) -> dict:
+        r"""
+        Moves the `has_calibration` field from the `WorkflowExecution` document to
+        the corresponding `DataGeneration` document.
+
+        >>> from nmdc_schema.migrators.adapters.dictionary_adapter import DictionaryAdapter
+        >>> db = {
+        ...   'workflow_execution_set': [
+        ...     {'id': 'nmdc:wfx1', 'has_calibration': 'nmdc:dobj-13-abc123', 'was_informed_by': 'nmdc:dgen1'},
+        ...     {'id': 'nmdc:wfx2', 'has_calibration': 'false', 'was_informed_by': 'nmdc:dgen2'}
+        ...   ],
+        ...   'data_generation_set': [
+        ...     {'id': 'nmdc:dgen1'},
+        ...     {'id': 'nmdc:dgen2'}
+        ...   ],
+        ...   'data_object_set': [
+        ...     {'id': 'nmdc:dobj-13-abc123'}
+        ...   ],
+        ...   'calibration_set': [
+        ...     {'id': 'nmdc:calib1', 'calibration_object': 'nmdc:dobj-13-abc123'}
+        ...   ]
+        ... }
+        >>> a = DictionaryAdapter(database=db)
+        >>> m = Migrator(adapter=a)
+        >>> workflow_execution_doc = {'id': 'nmdc:wfx1', 'has_calibration': 'nmdc:dobj-13-abc123', 'was_informed_by': 'nmdc:dgen1'}
+        >>> m.store_and_remove_calibrations(workflow_execution_doc)
+        {'id': 'nmdc:wfx1', 'was_informed_by': 'nmdc:dgen1'}
+        >>> workflow_execution_doc = {'id': 'nmdc:wfx2', 'has_calibration': 'false', 'was_informed_by': 'nmdc:dgen2'}
+        >>> m.store_and_remove_calibrations(workflow_execution_doc)
+        {'id': 'nmdc:wfx2', 'was_informed_by': 'nmdc:dgen2'}
+        >>> workflow_execution_doc = {'id': 'nmdc:wfx3', 'has_calibration': 'invalid', 'was_informed_by': 'nmdc:dgen3'}
+        >>> m.store_and_remove_calibrations(workflow_execution_doc)  # doctest: +IGNORE_EXCEPTION_DETAIL
+        Traceback (most recent call last):
+        ValueError: The 'has_calibration' value (invalid) in document (nmdc:wfx3) is not recognized
+        """
+
+        calibration_mapping = {} #create dictionary to store mappings
+
+        if "has_calibration" in workflow_execution_doc:
+            has_calibration_data_obj = workflow_execution_doc.get("has_calibration")
```

Comment:
I'd prefer this variable `has_calibration_data_obj` be renamed to `has_calibration_data_obj_id` (I appended `_id` to the end), since it contains an identifier as opposed to the identified thing.

Reply:
changed!